crates/katana/rpc/rpc/src/proxy_get_request.rs

```diff
@@ -0,0 +1,162 @@
+    }
+}
+
+/// Proxy `GET /path` requests to the specified RPC method calls.
+///
+/// # Request
+///
+/// The `GET /path` requests are modified into valid `POST` requests for
+/// calling the RPC method. This middleware adds appropriate headers to the
+/// request, and completely modifies the request `BODY`.
+///
+/// # Response
+///
+/// The response of the RPC method is stripped down to contain only the method's
+/// response, removing any RPC 2.0 spec logic regarding the response' body.
+#[derive(Debug, Clone)]
+pub struct DevnetProxy<S> {
+    inner: S,
+    path: Arc<str>,
+    method: Arc<str>,
+}
+
+impl<S> DevnetProxy<S> {
+    /// Creates a new [`DevnetProxy`].
+    ///
+    /// The request `GET /path` is redirected to the provided method.
+    /// Fails if the path does not start with `/`.
+    pub fn new(inner: S, path: &str, method: &str) -> Result<Self, RpcError> {
+        if !path.starts_with('/') {
+            return Err(RpcError::Custom(format!(
+                "DevnetProxy path must start with `/`, got: {}",
+                path
+            )));
+        }
+
+        Ok(Self { inner, path: Arc::from(path), method: Arc::from(method) })
+    }
+}
+
+impl<S> Service<Request<Body>> for DevnetProxy<S>
+where
+    S: Service<Request<Body>, Response = Response<Body>>,
+    S::Response: 'static,
+    S::Error: Into<Box<dyn Error + Send + Sync>> + 'static,
+    S::Future: Send + 'static,
+{
+    type Response = S::Response;
+    type Error = Box<dyn Error + Send + Sync + 'static>;
+    type Future =
+        Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>>;
+
+    #[inline]
+    fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        self.inner.poll_ready(cx).map_err(Into::into)
+    }
+
+    fn call(&mut self, mut req: Request<Body>) -> Self::Future {
+        let modify = self.path.as_ref() == req.uri() && req.method() == Method::GET;
+
+        // Proxy the request to the appropriate method call.
+        if modify {
+            // RPC methods are accessed with `POST`.
+            *req.method_mut() = Method::POST;
+            // Precautionary remove the URI.
+            *req.uri_mut() = Uri::from_static("/");
+
+            // Requests must have the following headers:
+            req.headers_mut().insert(CONTENT_TYPE, HeaderValue::from_static("application/json"));
+            req.headers_mut().insert(ACCEPT, HeaderValue::from_static("application/json"));
+
+            // Adjust the body to reflect the method call.
+            let raw_value = JsonRawValue::from_string("{\"address\":\"0x6677fe62ee39c7b07401f754138502bab7fac99d2d3c5d37df7d1c6fab10819\", \"age\":5, \"name\":\"somename\"}".to_string()).unwrap();
+            let param = Some(raw_value.as_ref());
+
+            let body = Body::from(
+                serde_json::to_string(&RequestSer::borrowed(&Id::Number(0), &self.method, param))
+                    .expect("Valid request; qed"),
```

Comment:
for GET methods, we can get the query params (eg localhost:5050/account_balance?contract_address=0x123) using `req.uri().query()` which would yield `contract_address=0x123`. then we cant construct the json body accordingly based on the method name.